sdk/app_cpu1/common/drv/pwm.c

```diff
@@ -83,6 +83,40 @@ void pwm_init(void)
     pwm_mux_set_all_pins(mux_config_data);
 }
 
+/*
+ * Sets the duty latching mode based off of the value in user_config.h
+ * DUTY RATIO UPDATE MODES:
+ * Mode 0: Update duty ratios at next timing manager trigger (default)
+ * Mode 1: Update duty ratios at next PWM carrier peak/valley
+ * Mode 2: Update duty ratios immediately (next FPGA clock rise)
+ */
+int pwm_set_duty_latching_mode(void)
+{
+    // Only allow PWM configuration changes when switching is off
+    if (pwm_is_enabled()) {
+        return FAILURE;
+    }
+
+    uint32_t config_reg_addr = PWM_BASE_ADDR + PWM_CONFIG_REG_OFFSET;
+
+#if USER_CONFIG_PWM_DUTY_LATCHING_MODE == 0
+    // Set slv_reg31[5:4] = 2'b00
+    Xil_Out32(config_reg_addr, (Xil_In32(config_reg_addr) & 0xFFFFFFCF));
+#elif USER_CONFIG_PWM_DUTY_LATCHING_MODE == 1
+    // Set slv_reg31[5:4] = 2'b01
+    Xil_Out32(config_reg_addr, (Xil_In32(config_reg_addr) | 0x00000010));
+    Xil_Out32(config_reg_addr, (Xil_In32(config_reg_addr) & 0xFFFFFFDF));
+#elif USER_CONFIG_PWM_DUTY_LATCHING_MODE == 2
+    // Set slv_reg31[5:4] = 2'b10
+    Xil_Out32(config_reg_addr, (Xil_In32(config_reg_addr) | 0x00000020));
+    Xil_Out32(config_reg_addr, (Xil_In32(config_reg_addr) & 0xFFFFFFEF));
+#else
+#error Invalid configuration for PWM Duty Latching Mode
+#endif
+
+    return SUCCESS;
+}
```

Comment:
Very nice.

However, let's allow the user to update this dynamically during run-time, i.e., change this function to accept an input and not be a compile-time setting.

In the user config file, have the user set an integer of `0` or `1` or `2`, and simply call this function with the user config value from the `init()` function. then, in here, you can check to make sure the input is valid etc.

Reply:
Ah! they already set the integer in the user config. so, just turn this function into if statements or switch or something

Reply:
@npetersen2 

Wait, so we do want to be able to update this at runtime? I don't see how just changing the `#if` to an `if ()` will make it configurable at runtime... wouldn't we need to make it an argument to the function to make this configurable at runtime?

Reply:
that is correct: we need to have an argument to the function so that it is configurable at runtime.